src/main/java/me/jellysquid/mods/lithium/mixin/entity/fast_piglin_brain/PiglinSpecificSensorMixin.java

```diff
@@ -0,0 +1,215 @@
+        boolean foundNemisis = false;
+        
+        for(LivingEntity currentLivingEntity : allVisibleMobs) {
+            // avoid many calls to isBaby() and just inline by hand to make sure
+            isAdult = !currentLivingEntity.isBaby();
+
+            /**
+             * To my knowledge, switches are faster than multiple if's.
+             * Also checking an int comparison in this case should be quicker than
+             * checking an instanceof, this should also be a bit more optimized for 
+             * the order at which things can be dectected to do early stopping in
+             * the switch statement due to how the fallthrough works.
+             * 
+             * To avoid more allocations, use casting as opposed to more variables
+             * and do so only when necessary
+             */
+            
+            switch(currentLivingEntity.getEntityId()){
+
+                // check if entity is a hoglin
+                case HOGLIN_ID : {
+                    if(isAdult) {
+                        visibleHoglinCount++;
+                        if(!foundHuntableHoglin && ((HoglinEntity)currentLivingEntity).canBeHunted()) {
+                            brain.remember(MemoryModuleType.NEAREST_VISIBLE_HUNTABLE_HOGLIN, Optional.of((HoglinEntity)currentLivingEntity));
+                            foundHuntableHoglin = true;
+                        }
+                    } else {
+                        if(!foundBabyHoglin) {
+                            brain.remember(MemoryModuleType.NEAREST_VISIBLE_BABY_HOGLIN, Optional.of((HoglinEntity)currentLivingEntity));
+                        }
+                    }
+                    break;
+                }
+
+                // check if entity is a piglin
+                case PIGLIN_ID : {
+                    if(isAdult) {
+                        visableAdultPiglins.add((PiglinEntity)currentLivingEntity);
+                    } else if(!foundBabyPiglin) {
+                        
+                    }
+                }
+            }
+
+            // stop iterating if all required targets were found
+            if(foundAttackablePlayer && foundBabyHoglin && foundBabyPiglin && foundHuntableHoglin && foundNemisis && foundPlayerHoldingGold)
+                return;
+        }
+
+        while(visibleMobsIterator.hasNext()) {
+            LivingEntity currentLivingEntity = (LivingEntity)visibleMobsIterator.next();
+
+            // Check hoglins
+            if (currentLivingEntity instanceof HoglinEntity) {
+                HoglinEntity hoglinEntity = (HoglinEntity)currentLivingEntity;
+                if (!nearestVisibleBabyHoglin.isPresent() && hoglinEntity.isBaby()) {
+                    nearestVisibleBabyHoglin = Optional.of(hoglinEntity);
+                } else if (hoglinEntity.isAdult()) {
+                    ++visibleHoglinCount;
+                    if (!nearestVisibleHuntableHoglin.isPresent() && hoglinEntity.canBeHunted()) {
```

Comment:
nearestVisibleHuntableHoglin is confusing, because it looks like it is using the first huntable hoglin. Or is the collection sorted by ascending distance?